src/hotspot/share/runtime/thread.hpp

```diff
@@ -970,6 +970,30 @@ class JavaThread: public Thread {
 
   static bool resize_all_jvmci_counters(int new_size);
 
+  void set_jvmci_reserved_oop0(oop value) {
+    _jvmci_reserved_oop0 = value;
+  }
+
+  oop get_jvmci_reserved_oop0() {
+    return _jvmci_reserved_oop0;
+  }
+
+  void set_jvmci_reserved0(jlong value) {
+    _jvmci_reserved0 = value;
+  }
+
+  jlong get_jvmci_reserved0() {
+    return _jvmci_reserved0;
+  }
+
+  void set_jvmci_reserved1(jlong value) {
+    _jvmci_reserved1 = value;
+  }
+
+  jlong get_jvmci_reserved1() {
+    return _jvmci_reserved1;
+  }
+
```

Comment:
This makes me reprise my wish that this code was in jvmciThread.hpp and you indirect from thread to there.  Can you do that?  There's already an RFE somewhere.

Reply:
I hadn't forgotten about that but I have been putting it off a bit.  I'll prioritize it.  The RFE is https://bugs.openjdk.java.net/browse/JDK-8137018